src/node/rpc/member_frontend.h

```diff
@@ -2058,6 +2062,371 @@ namespace ccf
+          {
+            auto root_at_read = ctx.tx.get_root_at_read_version();
+            if (!root_at_read.has_value())
+            {
+              return make_error(
+                HTTP_STATUS_INTERNAL_SERVER_ERROR,
+                ccf::errors::InternalError,
+                "Proposal failed to bind to state.");
+            }
+
+            // caller_identity.request_digest is set when getting the
+            // MemberSignatureAuthnIdentity identity. The proposal id is a
+            // digest of the root of the state tree at the read version and the
+            // request digest.
+            std::vector<uint8_t> acc(
+              root_at_read.value().h.begin(), root_at_read.value().h.end());
+            acc.insert(
+              acc.end(),
+              caller_identity.request_digest.begin(),
+              caller_identity.request_digest.end());
+            const crypto::Sha256Hash proposal_digest(acc);
+            proposal_id = proposal_digest.hex_str();
+          }
+          else
+          {
+            proposal_id = fmt::format(
+              "{:02x}", fmt::join(caller_identity.request_digest, ""));
+          }
+
+          js::JSAutoFreeRuntime rt;
+          js::JSAutoFreeCtx context(rt);
+          auto constitution = ctx.tx.ro(network.constitution)->get(0);
+          if (!constitution.has_value())
+          {
+            return make_error(
+              HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              ccf::errors::InternalError,
+              "No constitution is set - proposals cannot be evaluated");
+          }
+
+          auto code = constitution.value();
+          JSValue module = JS_Eval(
+            context,
+            code.c_str(),
+            code.size(),
+            "public:ccf.gov.constitution[0]",
+            JS_EVAL_TYPE_MODULE | JS_EVAL_FLAG_COMPILE_ONLY);
+
+          if (JS_IsException(module))
+          {
+            js::js_dump_error(context);
+            return make_error(
+              HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              ccf::errors::InternalError,
+              "Failed to compile constitution");
+          }
+
+          // Evaluate module
+          auto eval_val = JS_EvalFunction(context, module);
+          if (JS_IsException(eval_val))
+          {
+            js::js_dump_error(context);
+            return make_error(
+              HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              ccf::errors::InternalError,
+              "Failed to evaluate constitution");
+          }
+          JS_FreeValue(context, eval_val);
+
+          assert(JS_VALUE_GET_TAG(module) == JS_TAG_MODULE);
+          auto module_def = (JSModuleDef*)JS_VALUE_GET_PTR(module);
+          /*
+          if (JS_GetModuleExportEntriesCount(module_def) != 1)
+          {
+            throw std::runtime_error(
+              "Endpoint module exports more than one function");
+          }
+          */
+          auto export_func = JS_GetModuleExportEntry(context, module_def, 0);
```

Comment:
In your other PR you have `Context::function()`, I guess we want a similar one that accepts a function name. Should this be done in this PR or the other one?

Reply:
Yeah that's on the way separately.